lib/cosmos/gui/line_graph/line_graph_drawing.rb

```diff
@@ -244,19 +244,23 @@ def draw_x_label(dc, value, label)
     end # def draw_x_label_and_grid_line
 
     # Converts a x value into text with a max number of characters
-    def convert_x_value_to_text(value, max_characters)
-      if !@show_popup_x_y and @unix_epoch_x_values
-        if (value > 1 and value < 2147483647)
-          time = Time.at(value.to_f)
+    def convert_x_value_to_text(value, max_characters = @max_x_characters, full_date = false)
```

Comment:
I made max_characters default to the instance variable since it was never changed by the existing code. full_date is used by the popup code to display the full date in the case where they aren't using the formatted time item.